### PR TITLE
Upgrade 0.30

### PR DESCRIPTION
**Pull request by dorin-iancu**
- upgrade to elrond-wasm 0.30 version (also removing SCResult and #[payment] annotations)
- copy price aggregator proxy instead of importing it from another repo (less maintenance needed for future upgrades)
- use the new issue & set all roles function instead of having two endpoints for each token (lend/borrow)
- change pair swap call from async call to sync
- general code improvements for both performance and readability

### CODE REVIEW

File: savings-account/src/math.rs [hunk 1]
```diff
@@ -33,8 +32,7 @@ pub trait MathModule {
         borrowed_amount: &BigUint,
         total_pool_reserves: &BigUint,
     ) -> BigUint {
-        let bp = BigUint::from(BASE_PRECISION);
-        &(borrowed_amount * &bp) / total_pool_reserves
+        &(borrowed_amount * BASE_PRECISION) / total_pool_reserves
```
Comment (andrei-marinica) [inline savings-account/src/math.rs:35]: It's possible that the reference is superfluous, but not sure. Looks good anyway.

File: savings-account/src/ongoing_operation.rs [hunk 2]
```diff
@@ -80,24 +79,26 @@ pub trait OngoingOperationModule {
         gas_left > MIN_GAS_TO_SAVE_PROGRESS + extra_reserve_needed + operation_cost
     }
 
+    #[inline]
```
Comment (andrei-marinica) [inline savings-account/src/ongoing_operation.rs:82]: Normally no need for `#[inline]` unless you know for sure that there is a specific code size issue.

**Review by andrei-marinica — APPROVED** [review]

File: savings-account/src/math.rs (hunk 1 above)
Comment (dorin-iancu) [reply savings-account/src/math.rs]: You're right, it's no needed. Removed.

File: savings-account/src/ongoing_operation.rs (hunk 2 above)
Comment (dorin-iancu) [reply savings-account/src/ongoing_operation.rs:82]: I usually inline functions that are simply wrappers over other function calls, as the initial call only adds code size.

**Review by andrei-marinica — APPROVED** [review]

**Pull request merged**